src/lazy/encoder/binary/v1_1/flex_uint.rs

```diff
@@ -0,0 +1,229 @@
+                "write_large_u64() is only called for values whose encoded size is 9 or 10 bytes"
+            ),
+        }
+    }
+
+    pub fn value(&self) -> u64 {
+        self.value
+    }
+    pub fn size_in_bytes(&self) -> usize {
+        self.size_in_bytes
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::lazy::binary::immutable_buffer::ImmutableBuffer;
+    use crate::lazy::encoder::binary::v1_1::flex_uint::FlexUInt;
+    use crate::{IonError, IonResult};
+
+    const FLEX_UINT_TEST_CASES: &[(u64, &[u8])] = &[
+        (0u64, &[0b00000001]),
+        (1u64, &[0b00000011]),
+        (2u64, &[0b00000101]),
+        (3u64, &[0b00000111]),
+        (4u64, &[0b00001001]),
+        (5u64, &[0b00001011]),
+        (14u64, &[0b00011101]),
+        (63u64, &[0b01111111]),
+        (64u64, &[0b10000001]),
+        (127u64, &[0b11111111]),
+        (128u64, &[0b00000010, 0b00000010]),
+        (729u64, &[0b01100110, 0b00001011]),
+        (16383u64, &[0b11111110, 0b11111111]),
+        (16384u64, &[0b00000100, 0b00000000, 0b00000010]),
+        (2097151u64, &[0b11111100, 0b11111111, 0b11111111]),
+        (
+            2097152u64,
+            &[0b00001000, 0b00000000, 0b00000000, 0b00000010],
+        ),
+        (
+            268435455u64,
+            &[0b11111000, 0b11111111, 0b11111111, 0b11111111],
+        ),
+        (
+            268435456u64,
+            &[0b00010000, 0b00000000, 0b00000000, 0b00000000, 0b00000010],
+        ),
+        (
+            34359738368u64,
+            &[
+                0b00100000, 0b00000000, 0b00000000, 0b00000000, 0b00000000, 0b00000010,
+            ],
+        ),
+        (
+            4398046511104u64,
+            &[
+                0b01000000, 0b00000000, 0b00000000, 0b00000000, 0b00000000, 0b00000000, 0b00000010,
+            ],
+        ),
+        (
+            562949953421311u64,
+            &[
+                0b11000000, 0b11111111, 0b11111111, 0b11111111, 0b11111111, 0b11111111, 0b11111111,
+            ],
+        ),
+        (
+            562949953421312u64,
+            &[
+                0b10000000, 0b00000000, 0b00000000, 0b00000000, 0b00000000, 0b00000000, 0b00000000,
+                0b00000010,
+            ],
+        ),
+        (
+            72057594037927935u64,
+            &[
+                0b10000000, 0b11111111, 0b11111111, 0b11111111, 0b11111111, 0b11111111, 0b11111111,
+                0b11111111,
+            ],
+        ),
+        (
+            72057594037927936u64,
+            &[
+                0b00000000, 0b00000001, 0b00000000, 0b00000000, 0b00000000, 0b00000000, 0b00000000,
+                0b00000000, 0b00000010,
+            ],
+        ),
+        (
+            9223372036854775807u64,
+            &[
+                0b00000000, 0b11111111, 0b11111111, 0b11111111, 0b11111111, 0b11111111, 0b11111111,
+                0b11111111, 0b11111111,
+            ],
+        ),
+    ];
+
+    #[test]
+    fn decode_flex_uint() -> IonResult<()> {
```

Comment:
Can you add test cases for overpadded FlexUInts (such as `&[0b00000110, 0b00000000]`)—or are we just going to say that FlexUInts must be encoded with the smallest possible number of bytes?

Reply:
> or are we just going to say that FlexUInts must be encoded with the smallest possible number of bytes?

This is as of yet an unresolved question, though I think there are advocates for allowing padding. I'll add some tests for it.